posts/contributions-to-mathlib-from-lte.md

```diff
@@ -0,0 +1,50 @@
+---
+author: 'Riccardo Brasca'
+category: 'New in mathlib'
+date: 2021-10-19 11:04:21 UTC+02:00
+description: ''
+has_math: true
+link: ''
+slug: contributions-to-mathlib-from-lte
+tags: ''
+title: Contributions to mathlib from LTE
+type: text
+---
+When the [Liquid Tensor Experiment](https://github.com/leanprover-community/lean-liquid/) started,
+in December 2020, mathlib already
+[had](https://github.com/leanprover-community/mathlib/tree/c5009dd7140cf6ae53bf4ddeb57992eb10053b0b/src/analysis/normed_space)
+a decent theory of normed spaces. With this post I want to show how mathlib can benefit
+from projects like [LTE](https://github.com/leanprover-community/lean-liquid/), showing what we added
+to the theory of normed spaces in almost one year of work.
+
+Besides several small missing lemmas, we added the following notions.
+* `normed_group_hom`: we already had the operator norm, but no bundled hom between normed
+  groups. We introduced `normed_group_hom G H`, that it is itself a normed group. We also
+  introduced kernels and images of a normed groups hom.
+* `semi_normed_group`: a seminorm generalizes a norm by allowing nonzero vectors of zero
+  (semi)norm. This notion is needed in [LTE](https://github.com/leanprover-community/lean-liquid/),
+  and we introduced it in mathlib (providing a reasonable API). Since `normed_group` depends on
+  `metric_space` that in turn depends on `emetric_space`, we had first of all introduced
+  (extended) pseudo metric spaces. We also introduced `semi_normed_space` and similar related notions.
+* `normed_group_quotient`: the theory of quotients of (semi) normed groups was completely
+  missing. We now have a good API for it.
+* `normed_group_hom.completion`: similarly to `normed_group_quotient`, we add to develop
+  the theory of completion of normed groups and normed groups homs from scratch.
```

Comment:
This isn't true at all! This relies on the theory of completion of topological groups (which was part of the perfectoid spaces project).

Reply:
Ah yes, I meant there was nothing about completion of normed stuff.